pkg/k8sops/placement.go

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package k8sops
+
+import (
+	"fmt"
+
+	"github.com/m3db/m3cluster/generated/proto/placementpb"
+
+	myspec "github.com/m3db/m3db-operator/pkg/apis/m3dboperator/v1"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+// PlacementInstanceFromPod creates a new m3cluster placement instance given a
+// pod spec.
+func PlacementInstanceFromPod(cluster *myspec.M3DBCluster, pod *corev1.Pod) (*placementpb.Instance, error) {
+	isoGroup, ok := pod.ObjectMeta.Labels[_labelIsolationGroup]
+	if !ok {
+		return nil, fmt.Errorf("could not find label %s in %v", _labelIsolationGroup, pod.ObjectMeta.Labels)
+	}
+
+	setService := HeadlessServiceName(cluster.Name)
+	hostname := pod.Name + "." + setService
+
+	// TODO(schallert): dynamic zone, portname consts
+	instance := &placementpb.Instance{
+		Id:             pod.Name,
+		IsolationGroup: isoGroup,
+		Zone:           "embedded",
```

Comment:
can these be moved to `const`? Also don't we want `Zone` to be a different value per cluster?

Reply:
Yeah will do. And I think we'll have to have zone match whatever is in the dbnode config file, so maybe we can determine it from some field in the spec

Reply:
Yea I had a branch where I was thinking about reading the configuration map and verifying that the ports used in the configuration match the ports that were going to be used by the pods and services.  That could be expanded to match this field as well